src/Storages/MergeTree/ReplicatedMergeTreePartCheckThread.cpp

```diff
@@ -63,12 +72,47 @@ void ReplicatedMergeTreePartCheckThread::enqueuePart(const String & name, time_t
     if (parts_set.contains(name))
         return;
 
-    LOG_TRACE(log, "Enqueueing {} for check after after {}s", name, delay_to_check_seconds);
+    LOG_TRACE(log, "Enqueueing {} for check after {}s", name, delay_to_check_seconds);
     parts_queue.emplace_back(name, std::chrono::steady_clock::now() + std::chrono::seconds(delay_to_check_seconds));
     parts_set.insert(name);
     task->schedule();
 }
 
+void ReplicatedMergeTreePartCheckThread::enqueueBackgroundPartCheck(std::chrono::milliseconds delay_between_checks)
+{
+    auto background_part_check_time_to_total_time_ratio = storage.getSettings()->background_part_check_time_to_total_time_ratio;
+    if (background_part_check_time_to_total_time_ratio == .0f)
+        return;
+
+    /// safe guard
+    if (background_part_check_time_to_total_time_ratio > 1.0f)
+        background_part_check_time_to_total_time_ratio = 1.0f;
+    if (background_part_check_time_to_total_time_ratio < 1E-6f)
+        background_part_check_time_to_total_time_ratio = 1E-6f;
+
+    std::lock_guard lock(parts_mutex);
+
+    using namespace std::chrono;
+    milliseconds next_check_after_ms{last_check_duration};
+    if (last_check_duration.count())
+    {
+        const auto total_time_relative_to_last_check_duration = last_check_duration.count() / background_part_check_time_to_total_time_ratio;
+        if (total_time_relative_to_last_check_duration >= last_check_duration.count())
```

Comment:
How can total_time_relative_to_last_check_duration be < last_check_duration.count() if 
`total_time_relative_to_last_check_duration = last_check_duration.count() / background_part_check_time_to_total_time_ratio` where the ration is in (0, 1]?